### PR TITLE
framework/media: sync_up from rs6

### PR DESCRIPTION
**Pull request by jsdosa**
1. fix coverity issues for media/audio
2. optimize buffer header parsing

Signed-off-by: jsdosa <jsdosa@gmail.com>

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [23d737153445feab198091092d966a4f2ca6bde6] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [23d737153445feab198091092d966a4f2ca6bde6] - Code Rule Check Result:
framework/include/media/MediaUtils.h:49: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:VxO)
framework/include/media/MediaUtils.h:49: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/include/media/MediaUtils.h:49: ERROR: [SPC_M_OPR] need consistent spacing around '&' (ctx:VxW)
framework/include/media/MediaUtils.h:58: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:VxO)
framework/include/media/MediaUtils.h:58: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/include/media/MediaUtils.h:58: ERROR: [SPC_M_OPR] need consistent spacing around '&' (ctx:VxW)
framework/include/media/MediaUtils.h:68: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:VxO)
framework/include/media/MediaUtils.h:68: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/include/media/MediaUtils.h:78: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:VxO)
framework/include/media/MediaUtils.h:78: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)

**Comment by seinfra** [comment]
Target : [e9557de574935cd6fec77e4911db951bf86f841c] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [e9557de574935cd6fec77e4911db951bf86f841c] - Code Rule Check Result:
framework/include/media/MediaUtils.h:49: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:VxO)
framework/include/media/MediaUtils.h:49: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/include/media/MediaUtils.h:49: ERROR: [SPC_M_OPR] need consistent spacing around '&' (ctx:VxW)
framework/include/media/MediaUtils.h:58: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:VxO)
framework/include/media/MediaUtils.h:58: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/include/media/MediaUtils.h:58: ERROR: [SPC_M_OPR] need consistent spacing around '&' (ctx:VxW)
framework/include/media/MediaUtils.h:68: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:VxO)
framework/include/media/MediaUtils.h:68: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)
framework/include/media/MediaUtils.h:78: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:VxO)
framework/include/media/MediaUtils.h:78: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:OxV)

File: framework/src/media/utils/MediaUtils.cpp [hunk 1]
```diff
@@ -127,70 +185,181 @@ audio_type_t getAudioTypeFromMimeType(std::string &mimeType)
...
+			}
+			headPoint++;
+		}
+		if (isHeader && AAC_HEADER_LENGTH <= bufferSize - headPoint) {
+			header = (unsigned char *)malloc(sizeof(unsigned char) * (AAC_HEADER_LENGTH + 1));
+			if (header == NULL) {
+				medvdbg("malloc failed error\n");
+				return -1;
+			}
+			memcpy(header, buffer + headPoint, MP3_HEADER_LENGTH);
```
Comment (zhouxinhe) [inline framework/src/media/utils/MediaUtils.cpp]:
```suggestion
			memcpy(header, buffer + headPoint, AAC_HEADER_LENGTH );
```

**Review by tool3210 — APPROVED** [review]
LGTM

**Pull request closed without merging**